build_sysimg broken after #11640

Building recently from master (5444813fb8997b22cfa7ac78438d2f4798341152) produces a `sys.so` but no `sys.ji`. While the resulting binary will run from the command prompt, curiously it doesn't run properly when loaded via `node-julia`, which complains 

    System image file "/usr/local/lib/julia/sys.ji" not found

Attempting to build it manually fails also

    julia> build_sysimg(default_sysimg_path, "x86-64", force=true)
    INFO: Building inference0.o...
    /usr/local/bin/julia -C x86-64 --build /usr/local/bin/../lib/julia/inference0 coreimg.jl
    julia: unknown option `--build`
    ERROR: failed process: Process(`/usr/local/bin/julia -C x86-64 --build /usr/local/bin/../lib/julia/inference0 coreimg.jl`, ProcessExited(1)) [1]
     in run at ./process.jl:490
     in anonymous at /usr/local/share/julia/build_sysimg.jl:55
     in cd at ./file.jl:22
     in build_sysimg at /usr/local/share/julia/build_sysimg.jl:27
     in build_sysimg at /usr/local/share/julia/build_sysimg.jl:12

I don't know if this is related, but trying to ```make``` Julia fails for me with
``` >make
 JULIA usr/lib/julia/inference0.o
/bin/sh: line 1: 998 Illegal instruction (core dumped) /home/.../julia/usr/bin/julia -C native --output-o /home/.../julia/usr/lib/julia/inference0.o -f coreimg.jl
Makefile:176: recipe for target ´/home/.../julia/usr/lib/julia/inference0.o´ failed
make[1]: *** [/home/.../julia/usr/ib/julia/inference0.o] Error 132
Makefile:64: recipe for target ´julia-sysimg´ failed
make: *** [julia-sysimg] Error 2
```

Any ideas on how to fix this? Should the above PR help?

--- a/contrib/build_sysimg.jl
+++ b/contrib/build_sysimg.jl
@@ -51,19 +51,19 @@ function build_sysimg(sysimg_path=default_sysimg_path, cpu_target="native", user
             # Start by building inference0.{ji,o}
             inference0_path = joinpath(dirname(sysimg_path), "inference0")
             info("Building inference0.o...")
-            println("$julia -C $cpu_target --build $inference0_path coreimg.jl")
-            run(`$julia -C $cpu_target --build $inference0_path coreimg.jl`)
+            println("$julia -C $cpu_target --output-ji $inference0_path.ji --output-o $inference0_path.o coreimg.jl")
+            run(`$julia -C $cpu_target --output-ji $inference0_path.ji --output-o $inference0_path.o coreimg.jl`)
 
             # Bootstrap off off that to create inference.{ji,o}
             inference_path = joinpath(dirname(sysimg_path), "inference")
             info("Building inference.o...")
-            println("$julia -C $cpu_target --build $inference_path coreimg.jl")
-            run(`$julia -C $cpu_target --build $inference_path coreimg.jl`)
+            println("$julia -C $cpu_target --output-ji $inference_path.ji --output-o $inference_path.o coreimg.jl")
+            run(`$julia -C $cpu_target --output-ji $inference_path.ji --output-o $inference_path.o coreimg.jl`)
 
             # Bootstrap off off that to create sys.{ji,o}
             info("Building sys.o...")
-            println("$julia -C $cpu_target --build $sysimg_path -J $inference_path.ji -f sysimg.jl")
-            run(`$julia -C $cpu_target --build $sysimg_path -J $inference_path.ji -f sysimg.jl`)
+            println("$julia -C $cpu_target --output-ji $sysimg_path.ji --output-o $sysimg_path.o  -J $inference_path.ji -f sysimg.jl")
+            run(`$julia -C $cpu_target --output-ji $sysimg_path.ji --output-o $sysimg_path.o -J $inference_path.ji -f sysimg.jl`)
 
             if ld != nothing
                 link_sysimg(sysimg_path, ld)
@@ -72,7 +72,11 @@ function build_sysimg(sysimg_path=default_sysimg_path, cpu_target="native", user
             end
 
             if !Base.samefile("$default_sysimg_path.ji", "$sysimg_path.ji")
-                info("To run Julia with this image loaded, run: julia -J $sysimg_path.ji")
+                if Base.isfile("$sysimg_path.$(Libdl.dlext)")
+                    info("To run Julia with this image loaded, run: julia -J $sysimg_path.$(Libdl.dlext)")
+                else
+                    info("To run Julia with this image loaded, run: julia -J $sysimg_path.ji")
+                end
             else
                 info("Julia will automatically load this system image at next startup")
             end
@@ -166,7 +170,7 @@ if !isinteractive()
         println(" Example:")
         println("   build_sysimg.jl /usr/local/lib/julia/sys core2 ~/my_usrimg.jl --force")
         println()
-        println(" Running this script with no arguments is equivalent to calling it via")
+        println(" Running this script with no arguments is equivalent to:")
         println("   build_sysimg.jl $(default_sysimg_path) native")
         return 0
     end

